src/schemas/Program/index.js

```diff
@@ -352,10 +351,19 @@ const resolvers = {
 				: resolvePublicSingleProgram(shortName);
 		},
 
-		programs: async (obj, args, context, info) => {
+		programs: async (obj, args, context) => {
 			const { egoToken } = context;
-			return resolvePrivateProgramList(egoToken);
+			const { dataCenter } = args;
+
+			const programs = await resolvePrivateProgramList(egoToken);
+
+			const filteredPrograms = !dataCenter
+				? programs
+				: programs.filter((program) => program.dataCenter?.shortName === dataCenter);
```

Comment:
can you invert the check here? simpler to read with the positive condition:


```ts
			const filteredPrograms = dataCenter
				? programs.filter((program) => program.dataCenter?.shortName === dataCenter)
				: programs;
```

Reply:
Np sorry about that, updated